packages/react-native-gesture-handler/src/v3/createNativeWrapper.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useImperativeHandle, useRef } from 'react';
+
+import { NativeWrapperProps } from './hooks/utils';
+import { useNativeGesture } from './hooks/gestures';
+import { NativeDetector } from './detectors/NativeDetector';
+import type { NativeWrapperProperties } from './types/NativeWrapperType';
+import { NativeGesture } from './hooks/gestures/native/useNativeGesture';
+import { DetectorType, InterceptingGestureDetector } from './detectors';
+import { VirtualDetector } from './detectors/VirtualDetector/VirtualDetector';
+
+export type ComponentWrapperRef<P> = {
+  componentRef?: React.ComponentType<P>;
+  gestureRef?: NativeGesture;
+};
+
+export default function createNativeWrapper<P>(
+  Component: React.ComponentType<P>,
+  config: Readonly<NativeWrapperProperties> = {},
+  detectorType: DetectorType = DetectorType.Native
+) {
+  const ComponentWrapper = (
+    props: P & NativeWrapperProperties & { ref?: React.RefObject<unknown> }
+  ) => {
+    // Filter out props that should be passed to gesture handler wrapper
+    const { gestureHandlerProps, childProps } = Object.keys(props).reduce(
+      (res, key) => {
+        // @ts-ignore TS being overly protective with it's types, see https://github.com/microsoft/TypeScript/issues/26255#issuecomment-458013731 for more info
+        if (NativeWrapperProps.has(key)) {
+          // @ts-ignore FIXME(TS)
+          res.gestureHandlerProps[key] = props[key];
+        } else {
+          // @ts-ignore FIXME(TS)
+          res.childProps[key] = props[key];
+        }
+        return res;
+      },
+      {
+        gestureHandlerProps: { ...config }, // Watch out not to modify config
+        childProps: {
+          enabled: props.enabled,
+          hitSlop: props.hitSlop,
```

Comment:
The previous implementation was also passing `testID`, so why aren't we doing this here?